exp-player/addon/mixins/exp-player-route.js

```diff
@@ -0,0 +1,43 @@
+import Ember from 'ember';
+
+export default Ember.Mixin.create({
+    _experiment: null,
+    _session: null,
+
+    currentUser: Ember.inject.service(),
+
+    _getExperiment(params) { // jshint ignore: line
+
+    },
+    _getSession(params, experiment) { // jshint ignore: line
+
+    },
+    model(params) {
+        // While a little gross, this ensures all the criteria for participation
+        // are met before the route resolves. This has the benefit that the route's
+        // loading state is active until all of this is complete.
+        return this._getExperiment(params).then((experiment) => {
```

Comment:
Lower priority than sorting out profiles vs accounts, but...

You may be able to simplify some of this using RSVP.hash (which doesn't resolve until all constituent promises are resolved, thus ensuring the model hook doesn't finish). 

While this blog's advice tends to be of very mixed quality, here is an example that might be useful. (incl using `setProperties`)
http://emberigniter.com/load-multiple-models-single-route/#rsvp-hash-to-the-rescue:7f3cf4d8f2178a468d49396e07502eab